packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanDetail/KeyValuesTable.test.js

```diff
@@ -138,7 +138,7 @@ describe('<KeyValuesTable>', () => {
     expect(el.length).toBe(data.length);
     el.forEach((valueDiv, i) => {
       if (data[i].key !== 'jsonkey') {
-        expect(valueDiv.html()).toMatch(`"${data[i].value}"`);
+        expect(valueDiv.html()).toMatch(data[i].value.toString());
```

Comment:
Could we change the data table to contain the exact expectation instead of relying on toString()?

Reply:
I did try to make this test more meaningful by _comparing against the actual text_ instead of _matching against part of the DOM_, but it did not work:
```
// ----------- this works
valueDiv.html()
// =>  "<div class="ub-inline-block"><div class="json-markup"><span class="json-markup-string">some text</span></div></div>"

valueDiv.text()
// => "some text"

// ----------- this not
valueDiv.html()
// =>  "<div class="ub-inline-block"><div><div class="json-markup"><span class="json-markup-number">some text</span></div></div></div>"

valueDiv.text()
// => ""

```
I know it is somehow related to the shallow representation of the DOM inside this test, but I am really not familiar with react and its test frameworks, so I am at a dead end.
If you can help me get `"some text"` in the second example, I would implement the test change. Apart from that, testing with a regular expression against the DOM is just as bad as the call to `toString()`, if you ask me, so the latter itself doesn't really degrade the overall quality of the test.

Reply:
sorry, maybe I didn't explain clearly. The test is data driven from the `data` table. I suggest adding the expected values to that table, e.g. `{ key: 'numeric', value: 123456789, expected: "123456789" }` and comparing with those, instead of `value.toString()` (which is not as obvious in terms of what it would produce).